src/ChartBackground/index.js

```diff
@@ -14,149 +14,159 @@ import basicRender from '../canvas/basicRender'
+    if (!includes(groupedKeys, 'y')) return undefined
+    if (yShowTicks === false) return undefined
+    const defaultOffset = theme.axisTickFontSize * (theme.lineHeight - 1)
+    const {backgroundOffset, plotRect, yScale, yTickOffset = defaultOffset} = this.props
+    return map(yTicks, d => ({
+      type: 'text',
+      text: d.text,
+      x: sum([plotRect.x, -backgroundOffset, -yTickOffset]),
+      y: yScale(d.value),
+      textAlign: 'right',
+      textBaseline: 'middle',
+      fill: theme.textFill,
+      font: `${theme.axisTickFontSize}px ${theme.fontFamilyMono}`,
+    }))
+  }
 
-function getLabelText(props, key) {
-  const text = props[`${key}Name`] || props[key]
-  if (text) return text
-  const layer = findLast(props.layers, d => d[`${key}Name`] || d[key])
-  if (layer) return layer[`${key}Name`] || layer[key]
-  return undefined
-}
+  getBackgroundRenderData() {
+    const background = this.getBackground()
+    const xTicks = this.props.xTicks || getTicks(this.props, 'x')
+    const yTicks = this.props.yTicks || getTicks(this.props, 'y')
+    const xGuides = this.getXGuides(xTicks)
+    const yGuides = this.getYGuides(yTicks)
+    const xText = this.getXText(xTicks)
+    const yText = this.getYText(yTicks)
+    const thickXGuide = this.getXGuides(filter(xTicks, d => d.value === 0), true)
+    const thickYGuide = this.getYGuides(filter(yTicks, d => d.value === 0), true)
+    return flatten(compact([background, xGuides, yGuides, thickXGuide, thickYGuide, xText, yText]))
+  }
 
-export default function ChartBackground(props) {
-  const {height, plotRect, theme, width, xShowLabel, yShowLabel} = props
-  return (
-    <React.Fragment>
-      <Canvas
-        height={height}
-        plotRect={plotRect}
-        render={basicRender}
-        renderData={getBackgroundRenderData(props)}
-        theme={theme}
-        width={width}
-      />
-      {yShowLabel && (
-        <LeftLabel plotRect={plotRect} text={getLabelText(props, 'y')} theme={theme} />
-      )}
-      {xShowLabel && (
-        <BottomLabel plotRect={plotRect} text={getLabelText(props, 'x')} theme={theme} />
-      )}
-    </React.Fragment>
-  )
-}
+  getLabelText(key) {
+    const {layers} = this.props
+    const keyName = `${key}Name`
+    const text = this.props[keyName] || this.props[key]
+    if (text) return text
+    const layer = findLast(layers, d => d[keyName] || d[key])
+    if (layer) return layer[keyName] || layer[key]
+    return undefined
+  }
 
-ChartBackground.propTypes = {
-  height: PropTypes.number,
-  onUpdate: PropTypes.func,
-  plotRect: CustomPropTypes.plotRect.isRequired,
-  theme: CustomPropTypes.theme.isRequired,
-  width: PropTypes.number,
-  x: PropTypes.string,
-  xName: PropTypes.string,
-  xShowLabel: PropTypes.bool,
-  y: PropTypes.string,
-  yName: PropTypes.string,
-  yShowLabel: PropTypes.bool,
+  render() {
+    const {height, plotRect, theme, width, xShowLabel, yShowLabel} = this.props
+    const yText = this.getLabelText('y')
```

Comment:
What's the reasoning behind moving this out here? Isn't it more efficient to only call these functions if we need to?

Reply:
You're right, thanks.